_posts/2023-10-06-k8s-comparison-azure-aws.md

```diff
@@ -0,0 +1,264 @@
+
+## Architectural components for Kubernetes setup
+
+From an architectural perspective, both setups are very similar and highly specific to the cloud of choice at the same time.
+
+Let's start with the Kubernetes control plane. 
+This is very similar for both AWS and Azure as these are both fully managed services that are deployed and maintained by the provider.
+In both cases, no access to the actual control plane nodes is available, nor are they even visible. 
+The etcd is not exposed to the customer in either of the provider's managed services. 
+They both support public (default) and private control plane deployments. 
+
+W.r.t. the data plane, the setups are similar as well. 
+On AWS, the data plane runs on AWS EC2 nodes using Auto-Scaling Groups (ASG).
+On Azure, it runs on top of Azure Virtual Machines using VM Scale-Sets.
+Out of the box, these nodes are deployed into their equivalent of a private network, VPC or VNET, and a group of nodes is assigned to a subnet of that network. 
+
+AWS has the option to not run directly on top of EC2 instances, but use AWS Fargate instead. 
+This service allows EKS to deploy containers without having to spin up and maintain nodes.
+A similar option is available on Azure, namely Azure Container Instances. 
+Due to some caveats with the integrations which cause bad portability, these options were excluded from this comparison.
+
+## How managed is the managed service: out of the box support
+
+### Auto-scaling cluster nodes
+
+One of the big advantages of using a public cloud provider is that your "hardware" footprint can easily scale up and down, the so-called elasticity of the cloud. 
+It's such a big selling point for the public cloud that AWS used it for their service naming scheme. 
+
+It's no wonder that it's also a key feature for the managed Kubernetes offerings for the different cloud providers.
+Both AWS and Azure support using the standard Kubernetes autoscaler project to automatically provision and destroy the additional capacity for the data plane.
+However, their support and documentation are significantly different. 
+
+Enabling autoscaling on AKS is as easy as enabling the option during cluster creation.
+When adding the default (or any other) nodepool, the autoscaling option can be enabled. 
+This option will make sure that all required components, resources and configurations are created to support autoscaling of the data plane nodes.
+This includes installing the cluster auto scaler component into the cluster and setting up the required roles and rights for it to manage the node pool(s).
+The [Azure AKS documentation](https://learn.microsoft.com/en-us/azure/aks/cluster-autoscaler){:target="_blank" rel="noopener noreferrer"} shows how it can be enabled.
+There is some configuration possible, but the cluster auto-scaling configuration will be managed by Azure and isn't available to the user to change.
+
+On EKS, the journey is a bit more hands-on.
```

Comment:
Misschien hier ook iets van Fargate vermelden? Not sure

Reply:
Dit haal ik al aan ter hoogte van de intro 

`AWS has the option to not run directly on top of EC2 instances, but use AWS Fargate instead. `